### ISSUE TITLE
Direct Access to ManyToMany relation

### ISSUE THREAD

**Issue by awaitJosh** [OData/WebApi#1706]
I want to provide a possibility to my customer to get direct access to a manytomany-relation regardless the crossTable.

I have the following model:

`public class Employee
 {
     public Guid Id {get;set;}
     ....
     public ICollection<EmployeePosition> EmployeePositions {get;set;}
 }`

`public class Position
 {
     public Guid Id {get;set;}
     ....
     public ICollection<EmployeePosition> EmployeePositions {get;set;}
 }`

`public class EmployeePosition
 {
     public Guid Employee_Id {get;set;}
     public Guid Position_Id {get;set;}
 }`

Obviously I get problems if I try to implement a route like employee(someGuid)/positions/$ref. But I actually want something similar because I don´t want the client to handle the crosstable - can you give me any idea what to do here?

Thanks in advance, Josh

**Comment by KanishManuja-MS** [comment]
@awaitJosh I am not sure if I understood all your needs but do you think should simplify your model to have positions as a contained entity for a request like the one you mentioned.

**Comment by awaitJosh** [comment]
@KanishManuja-MS thank you for your answer! Sometimes its not easy to express myself in english.. I´ve edited my question (and I fixed the poor markup) maybe its more clear now..? Or is this simply not valid with OData? I´m still learning ;)

**Comment by awaitJosh** [comment]
Hi @KanishManuja-MS,
sorry if I annoy you.. But can you please give me, at least, a little hint..?

Kind regards

**Comment by KanishManuja-MS** [comment]
@awaitJosh Not at all, sorry for the delay in getting back to this. You will not be able to accomplish this without changing your model.

**Comment by awaitJosh** [comment]
@KanishManuja-MS thank you very much for getting back to this and for your patience! I´ve edited my question ones again because of the little spark of hope that I didn´t explain well. 

Changing the model would be very, very hard - maybe I´ve got something completely wrong but I thought that one magic OData-thing is, that the client can consume my database via webApi - I have a solution with about 25 tables, and many relationships so I simply can´t change the model.

Apart of a client-side evaluation of the post/tag relationship - is there any other way to solve this issue, as long as ef core can´t handle many-to-many relations same as EF6 ?

Kind regards and thank you again
